### ISSUE TITLE
Contents of div containing %sveltekit.body% no longer replaced in SvelteKit 2.5.5

### ISSUE THREAD

**Issue by theodorejb** [sveltejs/kit#12097]
### Describe the bug

I have an SPA mode app using adapter-static.

My `app.html` file contains the following inside the `<body>` tag:

```html
<div style="display: contents">
    <div class="loading-animation">
        <div></div>
        <div></div>
    </div>
    %sveltekit.body%
</div>
```

Up to SvelteKit 2.5.4, the `loading-animation` div was removed from the DOM when the SvelteKit app finishes loading. However, this no longer works in SvelteKit 2.5.5, and the loading animation continues to display on top of the loaded app.

### Reproduction

My top-level `+layout.ts` file contains:

```typescript
export const ssr = false;
export const prerender = false;
```

### System Info

```Shell
Windows 11, Firefox 124
```

### Severity

blocking an upgrade

### Additional Information

I suspect this issue may be caused by https://github.com/sveltejs/kit/pull/12050.

**Comment by eltigerchino** [comment]
I think this is the right behaviour since no SSR means no hydration, thus no repairing the DOM. It's easy enough to query the document for the loading element and remove it in an onMount callback.

cc: @Rich-Harris was disabling the hydration behaviour on SPA project considered a breaking change?

**Comment by theodorejb** [comment]
This is still an issue in the latest version of SvelteKit. It would be good to have a final decision if this is a bug which will be fixed, or intended behavior (anyone with an SPA-mode app needs to manually remove DOM siblings of `%sveltekit.body%`).

**Comment by lemmon** [comment]
+1 for this feature

**Comment by teemingc** [comment]
Tagging this as breaking change as I think it'd be best to standardise and document this behaviour. Whatever we decide is the correct behaviour is probably going to be a breaking change for someone.

**Comment by benmccann** [comment]
Even if hydration is enabled it's not immediately clear to me why you'd expect siblings of `%sveltekit.body%` to be removed upon hydration

Personally I would close this. I think there are cleaner ways to implement a loading spinner

**Comment by theodorejb** [comment]
Personally I don't care whether siblings of `%sveltekit.body%` are removed or not, but ideally it would work consistently regardless of whether hydration is enabled or using SPA client-side rendering.

**Comment by dummdidumm** [comment]
I am unable to reproduce this - the sibling div just stays around for me.

**Comment by teemingc** [comment]
Yeah, the behaviour seems consistent now. Siblings are never removed regardless of the `ssr` option value
